app/models/match.rb

```diff
@@ -19,23 +19,25 @@ def owner
     stage ? stage.owner : group.owner
   end
 
-  private
+  def winner
+    evaluate_winner
+    # This should maybe be cached or put into db
+  end
 
-  def stage_xor_group
-    errors.add(:stage_xor_group, 'Stage and Group missing or both present') unless stage.present? ^ group.present?
+  def group_match?
+    group.present?
   end
 
-  def evaluate_status
-    if score_team1 < score_team2
-      :team2_won
-    elsif score_team2 < score_team1
-      :team1_won
-    else
-      group_match? ? :undecided : :in_progress
-    end
+  private
+
+  def evaluate_winner
```

Comment:
Because I wanted to implement caching or writing it into database at first, then didn't manage to do so and decided to move that to another point in time.

Reply:
with caching (which is probably the better approach for this) the implementation will not change... therefore there's no need for two separate methods